### PR TITLE
Added async samples and refactored sync samples

### PR DESCRIPTION
**Pull request by mssfang**
- Added async samples and refactored sync samples
- Found an error that erroneous document count: 2  and valid document count: 0. It should be reversed. Created a follow-up issue for it at #7072

```
A batch of document statistics, document count: 2, erroneous document count: 2, transaction count: 2, valid document count: 0.
Document ID: 1
Recognized personal identifiable information entity: 555-55-5555, entity type: U.S. Social Security Number (SSN), entity subtype: N/A, offset: 10, length: 11, score: 0.85.
Document ID: 2
Recognized personal identifiable information entity: 4111 1111 1111 1111, entity type: Credit Card Number, entity subtype: N/A, offset: 10, length: 19, score: 0.85.
Batch of PII entities recognized.
```

fixes: #6937

### CODE REVIEW

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/AnalyzeSentiment.java [hunk 1]
```diff
@@ -20,8 +20,8 @@
     public static void main(String[] args) {
         // Instantiate a client that will be used to call the service.
         TextAnalyticsClient client = new TextAnalyticsClientBuilder()
-            .subscriptionKey("subscription-key")
-            .endpoint("https://servicename.cognitiveservices.azure.com/")
+            .subscriptionKey("<replace-with-your-text-analytics-key-here>")
+            .endpoint("<replace-with-your-text-analytics-endpoint-here>")
```
[the remark this answers is not in the document]
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/AnalyzeSentiment.java]: what about this 
```
.subscriptionKey("{subscription_key}")
.endpoint("https://{servicename}.cognitiveservices.azure.com/")
```

[the remark this answers is not in the document]
Comment (conniey) [reply sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/AnalyzeSentiment.java]: Works for me

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/AnalyzeSentimentAsync.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+package com.azure.ai.textanalytics;
+
+import com.azure.ai.textanalytics.models.TextSentiment;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Sample demonstrate how to analyze sentiment of a text input in asynchronously call.
```
Comment (conniey) [inline sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/AnalyzeSentimentAsync.java]:
```suggestion
 * Sample demonstrates how to asynchronously analyze the sentiment of an input text.
```

Comment (conniey) [reply sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/AnalyzeSentimentAsync.java]: Same with the others below. The sentences are worded oddly.

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/batch/AnalyzeSentimentBatchDocumentsAsync.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                    batchStatistics.getErroneousDocumentCount(),
+                    batchStatistics.getTransactionCount(),
+                    batchStatistics.getValidDocumentCount());
+
+                // Analyzed sentiment for each of document from a batch of documents
+                for (AnalyzeSentimentResult analyzeSentimentResult : analyzedBatchResult) {
+                    System.out.printf("Document ID: %s%n", analyzeSentimentResult.getId());
+                    final TextSentiment documentSentiment = analyzeSentimentResult.getDocumentSentiment();
+                    // Erroneous document
+                    if (documentSentiment == null) {
```
Comment (conniey) [inline sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/batch/AnalyzeSentimentBatchDocumentsAsync.java]: That seems like an odd way to check for error. I thought you had a `hasError()` method?

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/batch/AnalyzeSentimentBatchDocumentsAsync.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        List<TextDocumentInput> inputs = Arrays.asList(
+            new TextDocumentInput("1", "The hotel was dark and unclean. The restaurant had amazing gnocchi.", "en"),
+            new TextDocumentInput("2", "The restaurant had amazing gnocchi. The hotel was dark and unclean.", "en")
+        );
+
+        // Request options: show statistics and model version
+        final TextAnalyticsRequestOptions requestOptions = new TextAnalyticsRequestOptions().setShowStatistics(true);
+
+        // Analyzing batch sentiments
+        client.analyzeBatchSentimentWithResponse(inputs, requestOptions).subscribe(
```
Comment (conniey) [inline sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/batch/AnalyzeSentimentBatchDocumentsAsync.java]: Do you use the response headers? I think for samples, we should show use-cases we expect the general populace to use.... analzeBatchSentiment() would be my usual use-case.

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/batch/ExtractKeyPhrasesBatchDocuments.java [hunk 5]
```diff
@@ -37,21 +37,33 @@ public static void main(String[] args) {
...
         );
 
+        // Request options: show statistics and model version
         final TextAnalyticsRequestOptions requestOptions = new TextAnalyticsRequestOptions().setShowStatistics(true);
-        final DocumentResultCollection<ExtractKeyPhraseResult> detectedBatchResult = client.extractBatchKeyPhrasesWithResponse(inputs, requestOptions, Context.NONE).getValue();
-        System.out.printf("Model version: %s%n", detectedBatchResult.getModelVersion());
 
-        final TextDocumentBatchStatistics batchStatistics = detectedBatchResult.getStatistics();
+        // Extracting batch key phrases
+        final DocumentResultCollection<ExtractKeyPhraseResult> extractedBatchResult = client.extractBatchKeyPhrasesWithResponse(inputs, requestOptions, Context.NONE).getValue();
```
Comment (conniey) [inline sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/batch/ExtractKeyPhrasesBatchDocuments.java]: I'd reconsider the WithResponse calls. I thought we added this to encompass all use-cases, but I don't expect usual devs to care about the HTTP response.

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/HelloWorld.java [hunk 6]
```diff
@@ -20,23 +20,23 @@
...
-            .endpoint("https://servicename.cognitiveservices.azure.com/")
+            .subscriptionKey("<replace-with-your-text-analytics-key-here>")
+            .endpoint("<replace-with-your-text-analytics-endpoint-here>")
             .buildClient();
 
         // The text that need be analysed.
         String text = "hello world";
 
         final DetectLanguageResult detectLanguageResult = client.detectLanguage(text, "US");
         final DetectedLanguage detectedDocumentLanguage = detectLanguageResult.getPrimaryLanguage();
```
Comment (samvaity) [inline sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/HelloWorld.java]: nit: can we call this variable `detectedPrimaryLanguage`

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/batch/AnalyzeSentimentBatchDocumentsAsync.java (hunk 4 above)
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/batch/AnalyzeSentimentBatchDocumentsAsync.java]: Currently, we don't have `analyzeBatchSentiment(inputs, requestOptions)` API. So we will add one if needed

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/batch/ExtractKeyPhrasesBatchDocuments.java (hunk 5 above)
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/batch/ExtractKeyPhrasesBatchDocuments.java]: same to the async case that I mentioned above. Either create another API like `extractBatchKeyPhrases(inputs, requestOptions)` or not use extractBatchKeyPhrases(inputs) as example, I think we should show the user how to use requestOptions so I don't prefer the second way.

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/AnalyzeSentiment.java [hunk 7]
```diff
@@ -9,19 +9,19 @@
 import java.util.List;
 
 /**
- * Sample demonstrate how to analyze sentiment of a text input.
+ * Sample demonstrate how to analyze the sentiment of an input text.
```
Comment (conniey) [inline sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/AnalyzeSentiment.java]: demonstrate -> demonstrates

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/HelloWorld.java [hunk 8]
```diff
@@ -9,34 +9,33 @@
 import java.util.List;
 
 /**
- * Sample demonstrate how to detect language of a text input.
+ * Sample demonstrate how to detect the language of an input text.
```
Comment (conniey) [inline sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/HelloWorld.java]: demonstrate -> demonstrates

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/RecognizeEntities.java [hunk 9]
```diff
@@ -6,27 +6,27 @@
 import com.azure.ai.textanalytics.models.NamedEntity;
 
 /**
- * Sample demonstrate how to recognize entities of a text input.
+ * Sample demonstrate how to recognize the entities of an input text.
```
Comment (conniey) [inline sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/RecognizeEntities.java]: demonstrate -> demonstrates

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/AnalyzeSentiment.java (hunk 7 above)
Comment (conniey) [reply sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/AnalyzeSentiment.java]: There are a few others, but I stopped adding review comments for it.

**Pull request merged**